app/controllers/msb/MoneySourcesController.scala

```diff
@@ -64,17 +64,19 @@ class MoneySourcesController @Inject()(val authConnector: AuthConnector,
           case ValidForm(_, data: MoneySources) =>
             for {
               msb <- dataCacheConnector.fetch[MoneyServiceBusiness](MoneyServiceBusiness.key)
-              cache <- dataCacheConnector.save[MoneyServiceBusiness](MoneyServiceBusiness.key,
-                updateMoneySources(msb, data))
+              _ <- dataCacheConnector.save[MoneyServiceBusiness](MoneyServiceBusiness.key, updateMoneySources(msb, data))
+              maybeCache <- dataCacheConnector.fetchAll
+              cache <- Future.successful(maybeCache)
```

Comment:
Should we map over here instead of calling future.successful? There are examples in the code where we map over a cache fetch.